### DISCUSSION TITLE
mqtt password  characters long

### DISCUSSION THREAD

**Discussion by GTV1** [arendst/Tasmota#14599] [Q&A] [answered]
Hello, for all devices below 10. (e.g. 9.1) my mqtt password is 4 characters long. I have now updated a device to the latest version 10. When entering the MQTT password, it now asks me to enter at least 5 characters. Is this new ? and can I somehow go back to 4 characters? Otherwise I would have to assign a new password with 5 characters to 26 devices everywhere?!?!

**Comment by Jason2866** [comment, marked as answer]
Yes, minimum password length changed to 5 chars. To use shorter ones you have to change the Tasmota source code and compile yourself.

**Reply by GTV1** [reply]
Many Thanks for teh fast aswnser

**Reply by digiblur** [reply]
Make a second user for your new devices and abandon the old as you have time.

**Comment by wipeout666** [comment]
This check should exist when defining the password in the MQTT broker, not when entering the password in the device's config page.

**Comment by bfritscher** [comment]
using the web console sending: `MqttPassword 1234` worked for me

**Reply by joggs** [reply]
That saved me a day to reconfigure the mqtt server and reconfigure all clients. It should not be tasmotas responsibility as a client to force at least 5 chars. If anywhere it should be the server

**Reply by Yllelder** [reply]
Thank you very much.

I don't understand this stupidity of forcing me to use a password with a certain number of characters. I will know what password I want to use.

**Reply by s-hadinger** [reply]
Closing this discussion as it starts to become disrespectful to the Tasmota staff team. 

The answer has been provided. It is open source software, you can change this behavior at will. If you want super short passwords (and we don’t judge), you will need to compile your own version

**Reply by sfromis** [reply]
One reasoning is that it is many years ago that 4 characters would be seen as a "real" password. Of course, going to 5 does still not count much as "real security" anyway.
